src/Services/lessonsPlansScrapper.service.ts

```diff
@@ -0,0 +1,156 @@
+import { MajorLessonsPlanObject } from '../Types/majorLessonsPlanObject.type';
+import { LessonsPlanURLObject } from '../Types/lessonsPlanURLObject.type';
+import { LessonsPlanEntry } from '../Types/lessonsPlanEntry.type';
+import { getURLWithoutQuery } from '../utils/getURLWithoutQuery';
+import { transformCSVHeader } from '../utils/transformCSVHeader';
+import { transformCSVField } from '../utils/transformCSVField';
+import { checkIfFaculty } from '../utils/checkIfFaculty';
+import cheerio from 'cheerio';
+import Papa from 'papaparse';
+import * as _ from 'lodash';
+import axios from 'axios';
+
+const getDataFromCSV = async (
+    url: string,
+    type: 'main' | 'seminar' | 'faculty'
+): Promise<LessonsPlanEntry[]> => {
+    try {
+        const parserConfig = {
+            header: true,
+            dynamicTyping: true,
+            transformHeader: transformCSVHeader,
+            transform: transformCSVField,
+        };
+
+        const csv = await axios.get(
+            `https://inf.ug.edu.pl/plan/${url}&format=csv`
+        );
+
+        const results = Papa.parse(csv.data, parserConfig).data;
+
+        const withoutLastLine = results.splice(0, results.length - 1);
+
+        const finalResults: Array<LessonsPlanEntry> = withoutLastLine.map(
+            (result: any) => {
+                result.isFaculty = type === 'faculty';
+
+                if (type === 'seminar') {
+                    result.type = 'seminarium';
+                }
+
+                return result;
+            }
+        );
+
+        return finalResults;
+    } catch (error) {
+        return [];
+    }
+};
+
+const getAllUrls = async (
+    currentURL: string
+): Promise<LessonsPlanURLObject> => {
+    try {
+        const mainURL = 'https://inf.ug.edu.pl/plan/';
+        const planURLObject: LessonsPlanURLObject = { name: '', main: '' };
+
+        planURLObject.main = currentURL;
+
+        const { data } = await axios.get(mainURL + currentURL);
+        const $ = cheerio.load(data);
+        const name = $('div > h1').text();
+        planURLObject.name = name;
```

Comment:
create planURLObject here so that you don't have to alter it's attributes afterwards

Reply:
resolved